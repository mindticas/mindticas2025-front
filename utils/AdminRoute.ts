import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
