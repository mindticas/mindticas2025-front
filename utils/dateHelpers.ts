// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}