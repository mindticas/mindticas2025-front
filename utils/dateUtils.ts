import { DateTime } from "luxon";

interface DateTimeParts {
    date: string;
    time: string;
}

// * @returns Object with (YYYY-MM-DD) y time (HH:MM)

export const splitDateTimeFromISO = (
    isoString: string | null | undefined,
): DateTimeParts => {
    if (!isoString) return { date: '', time: '' };

    const dateObj = new Date(isoString);

    if (isNaN(dateObj.getTime())) {
        return { date: '', time: '' };
    }

    // format date (YYYY-MM-DD)
    const date = DateTime.fromJSDate(dateObj)
       .setZone('local')
        .toFormat('yyyy-MM-dd');

    // format time (HH:MM)
    const time = dateObj
        .toLocaleTimeString('es-ES', {
            hour: '2-digit',
            minute: '2-digit',
            hour12: false,
        })
        .replace(/\./g, '');

    return { date, time };
};
