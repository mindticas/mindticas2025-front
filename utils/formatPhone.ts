export default function formatPhone(phone: string): string {
    // Remove everything that is not digits
    const digits = phone.replace(/\D/g, '');
    // Applies format only if it has 10 digits
    if (digits.length === 10) {
        return `${digits.slice(0, 3)}-${digits.slice(3, 6)}-${digits.slice(6)}`;
    }
    return phone; // Returns the original if it doesn't match
}
