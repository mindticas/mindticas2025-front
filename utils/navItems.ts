import { BarChart, Calendar, Scissors, UserCog, Users } from 'lucide-react';

export const navItems = [
    { name: 'Citas', href: '/admin/', icon: Calendar },
    { name: 'Clientes', href: '/admin/clients', icon: Users },
    { name: 'Empleados', href: '/admin/employees', icon: UserCog },
    { name: 'Tratamientos', href: '/admin/treatments', icon: Scissors },
    { name: 'Reportes', href: '/admin/reports', icon: BarChart },
];
