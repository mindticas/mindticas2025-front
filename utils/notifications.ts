import { toaster } from '@/components/ui/toaster';

type ToastType = 'success' | 'error' | 'info' | 'warning';

interface ToastOption {
    duration?: number;
    title: string;
    description?: string;
}

const showToast = (type: ToastType, options: ToastOption) => {
    toaster.create({
        type,
        duration: options.duration || 5000,
        title: options.title,
        description: options.description,
    });
};

const TreatmentNotification = {
    createSuccess: () =>
        showToast('success', { title: 'Tratamiento creado con éxito' }),
    updateSuccess: () =>
        showToast('success', { title: 'Tratamiento actualizado con éxito' }),
    deleteSuccess: () =>
        showToast('success', { title: 'Tratamiento eliminado con éxito' }),
    createError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al crear el tratamiento',
        }),
    updateError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al actualizar el tratamiento0',
        }),
    deleteError: () =>
        showToast('error', { title: 'Error al eliminar el tratamiento' }),
    noChanges: () =>
        showToast('info', {
            title: 'No se detectaron cambios para actualizar',
        }),
};

export default TreatmentNotification;
