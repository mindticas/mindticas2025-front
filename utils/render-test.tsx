import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
