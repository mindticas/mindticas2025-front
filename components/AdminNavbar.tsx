'use client';
import { navItems } from '@/utils/navItems';
import { useBreakpointValue } from '@chakra-ui/react';
import { usePathname } from 'next/navigation';
import React, { useState } from 'react';
import AdminNavbarDesktop from './AdminNavbarDesktop';
import AdminNavbarMobile from './AdminNavbarMobile';

export default function AdminNavbar() {
    const pathname = usePathname();
    const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);
    const isDesktop = useBreakpointValue({ base: false, md: true });

    const handleToggleMenu = () => setIsMobileMenuOpen((prev) => !prev);

    return isDesktop ? (
        <AdminNavbarDesktop navItems={navItems} pathname={pathname} />
    ) : (
        <AdminNavbarMobile
            navItems={navItems}
            pathname={pathname}
            isMobileMenuOpen={isMobileMenuOpen}
            onToggleMenu={handleToggleMenu}
        />
    );
}
