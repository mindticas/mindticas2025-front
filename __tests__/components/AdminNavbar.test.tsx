import AdminNavbar from '@/pages/admin/AdminNavbar';
import { render } from '@/utils/render-test';

describe('AdminNavbar Snapshot', () => {
    it('should match the snapshot', () => {
        const { asFragment } = render(<AdminNavbar />);
        expect(asFragment()).toMatchSnapshot();
    });
});
