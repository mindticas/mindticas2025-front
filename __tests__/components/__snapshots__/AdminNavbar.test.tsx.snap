// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminNavbar Snapshot should match the snapshot 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-efzgkz"
  >
    <div
      class="css-lhqv6a"
    >
      <div
        class="css-70qvj9"
      >
        <a
          class="chakra-link css-1sa6w8m"
          href="/admin/citas"
        >
          <svg
            class="lucide lucide-scissors"
            fill="none"
            height="32"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="6"
              cy="6"
              r="3"
            />
            <path
              d="M8.12 8.12 12 12"
            />
            <path
              d="M20 4 8.12 15.88"
            />
            <circle
              cx="6"
              cy="18"
              r="3"
            />
            <path
              d="M14.8 14.8 20 20"
            />
          </svg>
          <p
            class="css-2jpnh5"
          >
            Elegangster Admin
          </p>
        </a>
      </div>
      <button
        aria-label="Abrir menÃº"
        class="chakra-button css-ld77z8"
        type="button"
      >
        <svg
          class="lucide lucide-menu"
          fill="none"
          height="24"
          stroke="white"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="4"
            x2="20"
            y1="12"
            y2="12"
          />
          <line
            x1="4"
            x2="20"
            y1="6"
            y2="6"
          />
          <line
            x1="4"
            x2="20"
            y1="18"
            y2="18"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
